presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddLocalExchanges.java

```diff
@@ -762,7 +763,7 @@ public PlanWithProperties visitUnion(UnionNode node, StreamPreferredProperties p
         public PlanWithProperties visitJoin(JoinNode node, StreamPreferredProperties parentPreferences)
         {
             PlanWithProperties probe;
-            if (isSpillEnabled(session) && isJoinSpillingEnabled(session)) {
+            if (isSpillEnabled(session) && isJoinSpillingEnabled(session) && !isNativeExecutionEnabled(session)) {
```

Comment:
Could you add a comment explaining why this is the case?  It seems to make it more confusing, and the comment would be helpful to separate the first two conditions (this is for some reason needed whenever we spill joins, stemming from a limitation in the Java implementation) from the second (native engine is implemented in such a way to avoid this limitation).

Reply:
@tdcmeehan Just saw this comment. I'll send a follow-up PR to address this.

Reply:
Created #22154